test/generators/OneOfSpec.js

```diff
@@ -1,89 +1,83 @@
-/**
- * Created by chetanv on 24/06/16.
- */
 
 import OneOf from '../../src/generators/OneOf';
 import { expect } from 'chai';
 import asyncFunction from '../test-helper/asyncFunction';
 
 describe('OneOf', function () {
   describe('#constructor', function () {
-    it('validates possible values', function () {
-      /* eslint-disable no-new */
-      function invalidValuesArray() {
-        new OneOf({}, 23);
-      }
-
-      function emptyValuesArray() {
-        new OneOf({}, []);
-      }
-
-      function validValuesArray() {
-        new OneOf({}, [1, 2, 3]);
-      }
-
-      /* eslint-enable no-new */
-
-      expect(invalidValuesArray).to.throw(Error);
-      expect(emptyValuesArray).to.throw(Error);
-      expect(validValuesArray).to.not.throw(Error);
+    it('can be created', function () {
+      const oneOf = new OneOf({});
+      expect(oneOf).to.be.instanceof(OneOf);
     });
   });
 
   describe('#generate', function () {
+    it('validates possible values', function () {
+      const oneOf = new OneOf({});
+      const invalidValuesArrayP = oneOf.generate(23);
+      const emptyValuesArrayP = oneOf.generate([]);
+      const validValuesArrayP = oneOf.generate([1, 2, 3]);
+
+      return Promise.all([
+        expect(invalidValuesArrayP).to.be.eventually.rejected,
+        expect(emptyValuesArrayP).to.be.eventually.rejected,
+        expect(validValuesArrayP).to.be.eventually.fulfilled,
+      ]);
+    });
+
     it('returns a promise', function () {
       const possibleValues = [1, 'two', 'III'];
-      const oneOf = new OneOf({}, possibleValues);
+      const oneOf = new OneOf({});
 
-      const valP = oneOf.generate();
+      const valP = oneOf.generate(possibleValues);
       expect(valP.then).to.be.a('function');
       return expect(valP).to.be.eventually.fulfilled;
     });
```

Comment:
@aexmachina, I think keeping oneOf async makes sense because it can then also support promises and async functions as well as other generators. e.g. consider a hypothetical case where a model attribute may hold one of two possible models, we can then use oneOf as:
```javascript
Factory.define('parent', Parent, {
  child: Factory.oneOf([ Factory.assoc('child1'), Factory.assoc('child2') ])
});
```
The `child` attribute will now be randomly associated with a `child1` or `child2` model.